cryptography/src/lib.rs

```diff
@@ -157,6 +155,27 @@ pub trait Digest:
     + Debug
     + Hash
 {
+    /// Attempts to read a digest from the provided buffer.
+    fn read_from<B: Buf>(buf: &mut B) -> Result<Self, Error> {
+        // Check if there are enough bytes in the buffer to read a digest.
+        let digest_len = size_of::<Self>();
+        if buf.remaining() < digest_len {
+            return Err(Error::InvalidDigestLength);
+        }
+
+        // If there are enough contiguous bytes in the buffer, use them directly.
+        let chunk = buf.chunk();
+        if chunk.len() >= digest_len {
+            let digest = Self::try_from(&chunk[..digest_len])?;
```

Comment:
Instead of using `DerefMut`, we actually call `TryFrom` in case Digest performs additional verification on top of just length checks.